src/cache/cache.rs

```diff
@@ -111,12 +112,13 @@ impl CacheRead {
     where
         T: Write,
     {
-        let mut file = self
+        let file = self
             .zip
             .by_name(name)
             .context("Failed to read object from cache entry")?;
-        io::copy(&mut file, to)?;
-        Ok(file.unix_mode())
+        let mode = file.unix_mode();
+        zstd::stream::copy_decode(file, to)?;
```

Comment:
How about returning an error if it's not the STORED compression method? After all, that means the file is not in the right format....